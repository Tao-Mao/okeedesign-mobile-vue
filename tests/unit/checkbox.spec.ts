import { mount, later } from '../index';
later;
describe('checkbox', () => {
  test('render checkbox selection status', async () => {
    const onClick = jest.fn();
    const wrapper = mount({
      template: `
        <o-checkbox-group v-model="checkbox" @click="onClick">
          <o-checkbox v-for="index in 4" :key="index" :value="index">
            {{ index }}
          </o-checkbox>
        </o-checkbox-group>`,
      data() {
        return {
          checkbox: [1],
        };
      },
      methods: {
        onClick() {
          onClick();
        },
      },
    });
    expect(wrapper).toMatchSnapshot();
    const checkboxs = wrapper.findAll('.o-checkbox');
    await checkboxs.at(1).trigger('click');
    await checkboxs.at(3).trigger('click');
    expect(wrapper).toMatchSnapshot();
    // expect(onClick).toBeCalledTimes(2);
  });
});
