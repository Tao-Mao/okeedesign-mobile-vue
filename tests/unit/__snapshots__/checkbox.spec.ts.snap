// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox render checkbox selection status 1`] = `
<div class="o-checkbox-group"><label class="o-checkbox o-checkbox--normal  o-checkbox--checked"><i class="o-icon o-icon--check-one o-checkbox__icon">test-file-stub</i><span class="o-checkbox__label">
            1
          </span><input type="checkbox" class="o-checkbox__input" value="1"></label><label class="o-checkbox o-checkbox--normal"><i class="o-icon o-icon--radio-uncheck o-checkbox__icon"></i><span class="o-checkbox__label">
            2
          </span><input type="checkbox" class="o-checkbox__input" value="2"></label><label class="o-checkbox o-checkbox--normal"><i class="o-icon o-icon--radio-uncheck o-checkbox__icon"></i><span class="o-checkbox__label">
            3
          </span><input type="checkbox" class="o-checkbox__input" value="3"></label><label class="o-checkbox o-checkbox--normal"><i class="o-icon o-icon--radio-uncheck o-checkbox__icon"></i><span class="o-checkbox__label">
            4
          </span><input type="checkbox" class="o-checkbox__input" value="4"></label></div>
`;

exports[`checkbox render checkbox selection status 2`] = `
<div class="o-checkbox-group"><label class="o-checkbox o-checkbox--normal  o-checkbox--checked"><i class="o-icon o-icon--check-one o-checkbox__icon">test-file-stub</i><span class="o-checkbox__label">
            1
          </span><input type="checkbox" class="o-checkbox__input" value="1"></label><label class="o-checkbox o-checkbox--normal  o-checkbox--checked"><i class="o-icon o-icon--check-one o-checkbox__icon">test-file-stub</i><span class="o-checkbox__label">
            2
          </span><input type="checkbox" class="o-checkbox__input" value="2"></label><label class="o-checkbox o-checkbox--normal"><i class="o-icon o-icon--radio-uncheck o-checkbox__icon"></i><span class="o-checkbox__label">
            3
          </span><input type="checkbox" class="o-checkbox__input" value="3"></label><label class="o-checkbox o-checkbox--normal  o-checkbox--checked"><i class="o-icon o-icon--check-one o-checkbox__icon">test-file-stub</i><span class="o-checkbox__label">
            4
          </span><input type="checkbox" class="o-checkbox__input" value="4"></label></div>
`;
